### PR TITLE
fix(a11y): TimePicker, DateInput a11y fixes

### PR DESCRIPTION
**Pull request by bvandrc**
Basing a11y structure off of this example: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/

### CODE REVIEW

File: packages/datetime/src/components/date-input/dateInput.tsx [hunk 1]
```diff
@@ -245,6 +246,7 @@ export const DateInput: React.FC<DateInputProps> = React.memo(function _DateInpu
 
     const inputRef = React.useRef<HTMLInputElement | null>(null);
     const popoverContentRef = React.useRef<HTMLDivElement | null>(null);
+    const popoverId = uniqueId("date-picker");
```
Comment (adidahiya) [inline packages/datetime/src/components/date-input/dateInput.tsx]: can you use `Utils.uniqueId()` instead? `import { Utils} from "@blueprintjs/core"`

File: packages/datetime/src/components/time-picker/timePicker.tsx [hunk 2]
```diff
@@ -197,6 +195,7 @@ export class TimePicker extends React.Component<TimePickerProps, TimePickerState
                 onKeyDown={this.getInputKeyDownHandler(unit)}
                 onKeyUp={this.getInputKeyUpHandler(unit)}
                 role={this.props.showArrowButtons ? "spinbutton" : undefined}
+                type="number"
```
Comment (adidahiya) [inline packages/datetime/src/components/time-picker/timePicker.tsx:197]: there's a reason why we did not use this attribute, as you can see this change broke the test suite. you'll need to revert this part

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: packages/datetime/src/components/time-picker/timePicker.tsx (hunk 2 above)
Comment (bvandrc) [reply packages/datetime/src/components/time-picker/timePicker.tsx:197]: I think that test can just be modified with this change. The test is essentially "if entry is not a number, display error", however setting `type=number` disallows non-number entry in the first place and solves this case / problem from the start.

Is there a reason to not have it be `type=number` when that's exactly what it is?

Tests updated: https://github.com/palantir/blueprint/commit/abe73d10ddb38d485631059a69a63bb6da102f21

Comment (bvandercar-vt) [reply packages/datetime/src/components/time-picker/timePicker.tsx:198]: Tests updated!  https://github.com/palantir/blueprint/pull/6568/commits/abe73d10ddb38d485631059a69a63bb6da102f21

Comment (bvandrc) [reply packages/datetime/src/components/time-picker/timePicker.tsx:197]: In fact, if we use `type=number`,  we can just use the `min` and `max` props instead of `aria-valuemin` and `aria-valuemax`

File: packages/core/src/components/section/section.tsx [hunk 3]
```diff
@@ -115,7 +115,7 @@ export interface SectionProps extends Props, Omit<HTMLDivProps, "title">, React.
     /**
      * Optional title renderer function. If provided, it is recommended to include a Blueprint `<H6>` element
      * as part of the title. The render function is supplied with `className` and `id` attributes which you must
-     * forward to the DOM.
+     * forward to the DOM, as well as the `children` which is the passed `title`.
```
Comment (adidahiya) [inline packages/core/src/components/section/section.tsx]: please remove this unrelated change from this PR. You can fix this JSDoc in a separate PR.

also the wording is a bit awkward, I would say this instead:

```suggestion
     * forward to the DOM. The `title` prop is also passed along to this renderer via `props.children`.
```

File: packages/datetime/src/components/time-picker/timePicker.tsx (hunk 2 above)
Comment (adidahiya) [reply packages/datetime/src/components/time-picker/timePicker.tsx:197]: I tried to dig up the original reason why we didn't use `type="number"` when the component was originally written 8 years ago I and couldn't find a compelling one. Perhaps the browser support was incomplete, or there wasn't good support for doing the style overrides you've added in your recent commits. I suppose we can switch to `type="number"` now, as it does seem like the right markup to use in the long run. I tested it out in the [docs preview build](https://output.circle-artifacts.com/output/job/2d566d4e-bcfd-4966-b768-af49e578f300/artifacts/0/packages/docs-app/dist/index.html#datetime/timepicker) and things seem to be working fine :+1:

File: packages/datetime/test/components/timePickerTests.tsx [hunk 4]
```diff
@@ -144,16 +144,27 @@ describe("<TimePicker>", () => {
         hourInput.value = "2";
         TestUtils.Simulate.change(hourInput);
         assert.strictEqual(hourInput.value, "2");
+        assert.isFalse(hourInput.classList.contains(CoreClasses.intentClass(Intent.DANGER)));
     });
 
-    it("allows invalid text entry, but shows visual indicator", () => {
+    it("disallows allows non-number text entry", () => {
```
Comment (adidahiya) [inline packages/datetime/test/components/timePickerTests.tsx]: I think there's a typo:

```suggestion
    it("disallows non-number text entry", () => {
```

**Review by adidahiya — CHANGES_REQUESTED** [review]

File: packages/core/src/components/section/section.tsx (hunk 3 above)
Comment (bvandrc) [reply packages/core/src/components/section/section.tsx]: Done

File: packages/datetime/test/components/timePickerTests.tsx (hunk 4 above)
Comment (bvandrc) [reply packages/datetime/test/components/timePickerTests.tsx]: Fixed.

**Review by adidahiya — APPROVED** [review]
changes lgtm, can you please run `yarn lint-fix` inside `packages/datetime/`?

**Comment by bvandrc** [comment]
> changes lgtm, can you please run `yarn lint-fix` inside `packages/datetime/`?

@adidahiya  Done.

**Review by adidahiya — APPROVED** [review]
